endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1929,3 +1934,69 @@ func TestSetTargeting(t *testing.T) {
 
 	}
 }
+
+func TestValidAmpResponseWhenRequestRejected(t *testing.T) {
+	nbr := openrtb3.NoBidReason(123)
+	reject := hookexecution.RejectError{
+		int(nbr),
+		hookexecution.HookID{"foobar", "foo"},
+		hooks.StageEntrypoint.String(),
+	}
+
+	testCases := []struct {
+		description         string
+		expectedAmpResponse AmpResponse
+		hookExecutor        hookexecution.HookStageExecutor
+	}{
+		{
+			"Assert correct AmpResponse when request rejected at entrypoint stage",
+			AmpResponse{Targeting: map[string]string{}},
+			rejectableHookExecutor{entrypointReject: &reject},
+		},
+	}
```

Comment:
There is no need to use a table driven test if there is only 1 condition. Will there be more conditions in the near future?

Reply:
We are going to extend this with new test cases in the following PRs, to check rejections made at other stages.

Reply:
Gotcha. Sounds good.